lib/sdotypes.c

```diff
@@ -993,6 +993,27 @@ sdo_string_t *sdo_string_alloc(void)
 	return (sdo_string_t *)sdo_alloc(sizeof(sdo_string_t));
 }
 
+/**
+ * Create sdo_string_t object with the given size.
+ * @return an allocated sdo_string_t object
+ */
+sdo_string_t *sdo_string_alloc_size(size_t byte_sz) {
+
+	sdo_string_t *s = (sdo_string_t *)sdo_alloc(sizeof(sdo_string_t));
+	
+	if (s == NULL)
+		return NULL;
+
+	if (byte_sz > 0) {
+		s->bytes = sdo_alloc(byte_sz * sizeof(char));
+		if (s->bytes == NULL)
+			return NULL;
+		s->byte_sz = byte_sz;
+		return s;
+	} else
```

Comment:
In case the if block is enclosed with {}, do the same with else block too.

} else {
    return s;
}

Reply:
Done.